client/src/app/containers/Game/DealAction.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import SendIcon from '@material-ui/icons/Send';
+import ShuffleIcon from '@material-ui/icons/Shuffle';
+
+import { ActionInputProps } from './types';
+import { useGame } from './useGame';
+
+const DealAction: React.FunctionComponent<ActionInputProps> = ({
+    isBlocking,
+}) => {
+    const { submitDealAction } = useGame();
+    const [numShuffles, setNumShuffles] = useState(0);
+    return (
+        <Grid item container spacing={2}>
+            <Button
+                disabled={!isBlocking}
+                variant='contained'
+                color='secondary'
+                endIcon={<ShuffleIcon />}
+                onClick={() => setNumShuffles(prev => prev + 1)}
```

Comment:
:mag: how the heck does this work?

Reply:
The `set*` function returned by `useState` most typically takes a value (the value to set the state to) but can also take a function denoting _how_ to set the state, which receives the previous value of said state as an arg. Could just as easily pass `numShuffles + 1` here if you prefer.

Reply:
Neat! it can take a val to set the state to, or the state transformation from the previous value! :til: